src/health_checker.h

```diff
@@ -38,9 +43,32 @@ class HealthChecker {
 
   const Configuration& config_;
   std::set<std::string> unhealthy_components_;
+  std::map<std::string, std::function<bool()>> component_callbacks_;
   mutable std::mutex mutex_;
 };
 
+// Registers a component and then unregisters when it goes out of
+// scope.
+class ScopedHealthCheckRegistration {
+ public:
+  ScopedHealthCheckRegistration(HealthChecker* health_checker,
+                                const std::string& component,
+                                std::function<bool()> callback)
+    : health_checker_(health_checker), component_(component) {
+    if (health_checker_ != nullptr) {
+      health_checker_->RegisterComponent(component_, callback);
+    }
+  }
+  ~ScopedHealthCheckRegistration() {
+    if (health_checker_ != nullptr) {
+      health_checker_->UnregisterComponent(component_);
+    }
+  }
+ private:
+  HealthChecker* health_checker_;
+  const std::string& component_;
```

Comment:
Uh, don't do this. If you store a reference to a temporary, you'll cause a crash. Just store a copy of the string.